### ISSUE TITLE
v4.0-alpha13 editor crash with project upgrade from v3.4.4.stable

### ISSUE THREAD

**Issue by blipk** [godotengine/godot#64262]
### Godot version

v4.0-alpha13

### System information

Arch AMD64

### Issue description

Ran the binary from the release here `https://downloads.tuxfamily.org/godotengine/4.0/alpha12/`
Project manager opens, import a project from v3.4.4 - says it will upgrade it and click ok.
As the editor is opening get the following log and crash:
```
    ~/opt  ./Godot_v4.0-alpha13_linux.64                                                                                                                                                              ✔ 
Godot Engine v4.0.alpha13.official.59dcddc45 - https://godotengine.org
ERROR: GENERAL - Message Id Number: 0 | Message Id Name: Loader Message
	loader_scanned_icd_add: Could not get 'vkCreateInstance' via 'vk_icdGetInstanceProcAddr' for ICD libGLX_nvidia.so.0
	Objects - 1
		Object[0] - VK_OBJECT_TYPE_INSTANCE, Handle 159703888
   at: _debug_messenger_callback (drivers/vulkan/vulkan_context.cpp:171)
Vulkan API 1.2.0 - Using Vulkan Device #0: AMD - AMD RADV RAVEN
 
Editing project: /home/user/Documents/Godot/Garden24 (::home::user::Documents::Godot::Garden24)
Godot Engine v4.0.alpha13.official.59dcddc45 - https://godotengine.org
ERROR: GENERAL - Message Id Number: 0 | Message Id Name: Loader Message
	loader_scanned_icd_add: Could not get 'vkCreateInstance' via 'vk_icdGetInstanceProcAddr' for ICD libGLX_nvidia.so.0
	Objects - 1
		Object[0] - VK_OBJECT_TYPE_INSTANCE, Handle 140205216
   at: _debug_messenger_callback (drivers/vulkan/vulkan_context.cpp:171)
Vulkan API 1.2.0 - Using Vulkan Device #0: AMD - AMD RADV RAVEN
    ~/opt                                                                                                                                                                                     ✔  15s  
SCRIPT ERROR: Parse Error: Unexpected "Identifier" in class body.
          at: GDScript::reload (res://util/Autoload.gd:3)
ERROR: Script does not inherit from Node: res://util/Autoload.gd.
   at: _create_autoload (editor/editor_autoload_settings.cpp:412)
SCRIPT ERROR: Parse Error: Unexpected "Identifier" in class body.
          at: GDScript::reload (res://util/Global.gd:5)
ERROR: Script does not inherit from Node: res://util/Global.gd.
   at: _create_autoload (editor/editor_autoload_settings.cpp:412)
WARNING: FBX file import is enabled, but no FBX2glTF path is configured. FBX files will not be imported.
     at: _editor_init (modules/gltf/register_types.cpp:92)
SCRIPT ERROR: Parse Error: Expected end of statement after variable declaration, found "Identifier" instead.
          at: GDScript::reload (res://player/AnimatedSprite.gd:3)
SCRIPT ERROR: Parse Error: Unexpected "Identifier" in class body.
          at: GDScript::reload (res://player/player.gd:9)
SCRIPT ERROR: Parse Error: Unexpected "Identifier" in class body.
          at: GDScript::reload (res://player/PlayerCrosshair.gd:3)
SCRIPT ERROR: Parse Error: Unexpected "Identifier" in class body.
          at: GDScript::reload (res://player/PlayerDebug.gd:3)
SCRIPT ERROR: Parse Error: Unexpected "Identifier" in class body.
          at: GDScript::reload (res://player/UILifeBar.gd:3)
SCRIPT ERROR: Parse Error: Could not find base class "KinematicBody2D".
          at: GDScript::reload (res://thing/thing.gd:0)
SCRIPT ERROR: Parse Error: Unexpected "Identifier" in class body.
          at: GDScript::reload (res://unused/Main.gd:5)
SCRIPT ERROR: Parse Error: Static function "get_scancode_string()" not found in base "OS".
          at: GDScript::reload (res://unused/MainLoopCustom.gd:22)
SCRIPT ERROR: Parse Error: Unexpected "Identifier" in class body.
          at: GDScript::reload (res://util/Autoload.gd:3)
SCRIPT ERROR: Parse Error: Unexpected "Identifier" in class body.
          at: GDScript::reload (res://util/Debug.gd:3)
SCRIPT ERROR: Parse Error: "PoolStringArray" was not found in the current scope.
          at: GDScript::reload (res://util/FileChooser.gd:39)
SCRIPT ERROR: Parse Error: Identifier "popup_exclusive" not declared in the current scope.
          at: GDScript::reload (res://util/FileChooser.gd:8)
SCRIPT ERROR: Parse Error: Function "PoolStringArray()" not found in base self.
          at: GDScript::reload (res://util/FileChooser.gd:15)
SCRIPT ERROR: Parse Error: Identifier "window_title" not declared in the current scope.
          at: GDScript::reload (res://util/FileChooser.gd:18)
SCRIPT ERROR: Parse Error: Identifier "MODE_SAVE_FILE" not declared in the current scope.
          at: GDScript::reload (res://util/FileChooser.gd:19)
SCRIPT ERROR: Parse Error: Invalid argument for "connect()" function: argument 2 should be Callable but is res://util/FileChooser.gd.
          at: GDScript::reload (res://util/FileChooser.gd:31)
SCRIPT ERROR: Parse Error: Invalid argument for "connect()" function: argument 3 should be Array but is String.
          at: GDScript::reload (res://util/FileChooser.gd:31)
SCRIPT ERROR: Parse Error: Invalid argument for "connect()" function: argument 2 should be Callable but is res://util/FileChooser.gd.
          at: GDScript::reload (res://util/FileChooser.gd:32)
SCRIPT ERROR: Parse Error: Invalid argument for "connect()" function: argument 3 should be Array but is String.
          at: GDScript::reload (res://util/FileChooser.gd:32)
SCRIPT ERROR: Parse Error: Invalid argument for "connect()" function: argument 2 should be Callable but is res://util/FileChooser.gd.
          at: GDScript::reload (res://util/FileChooser.gd:33)
SCRIPT ERROR: Parse Error: Invalid argument for "connect()" function: argument 3 should be Array but is String.
          at: GDScript::reload (res://util/FileChooser.gd:33)
SCRIPT ERROR: Parse Error: Identifier "BUTTON_RIGHT" not declared in the current scope.
          at: GDScript::reload (res://util/FileChooser.gd:45)
SCRIPT ERROR: Parse Error: Function "get_global_mouse_position()" not found in base self.
          at: GDScript::reload (res://util/FileChooser.gd:46)
SCRIPT ERROR: Parse Error: Unexpected "Identifier" in class body.
          at: GDScript::reload (res://util/Global.gd:5)
SCRIPT ERROR: Parse Error: Unexpected "Identifier" in class body.
          at: GDScript::reload (res://util/InputDisplay.gd:3)
SCRIPT ERROR: Parse Error: Unexpected "Identifier" in class body.
          at: GDScript::reload (res://util/ResPlayer.gd:5)
SCRIPT ERROR: Parse Error: Unexpected "Identifier" in class body.
          at: GDScript::reload (res://util/SceneTreeCustom.gd:15)
SCRIPT ERROR: Parse Error: Unexpected "Identifier" in class body.
          at: GDScript::reload (res://util/TimerMan.gd:4)
SCRIPT ERROR: Parse Error: Could not find base class "Reference".
          at: GDScript::reload (res://util/util.gd:0)
SCRIPT ERROR: Parse Error: Unexpected "Identifier" in class body.
          at: GDScript::reload (res://world/environment/background/ParallaxLayerClouds.gd:3)
SCRIPT ERROR: Parse Error: Unexpected "Identifier" in class body.
          at: GDScript::reload (res://world/environment/CloudLight2D.gd:3)
SCRIPT ERROR: Parse Error: Unexpected "Identifier" in class body.
          at: GDScript::reload (res://world/tilemaps/TileDebug.gd:8)
SCRIPT ERROR: Parse Error: Unexpected "Identifier" in class body.
          at: GDScript::reload (res://world/tilemaps/TileMap.gd:3)
SCRIPT ERROR: Parse Error: Could not resolve super class path "res://TileMap.gd".
          at: GDScript::reload (res://world/tilemaps/TileMapCoverHollow.gd:0)
SCRIPT ERROR: Parse Error: Unexpected "Identifier" in class body.
          at: GDScript::reload (res://world/tilemaps/TileSet.gd:1)
SCRIPT ERROR: Parse Error: Unexpected "Identifier" in class body.
          at: GDScript::reload (res://Main2D.gd:3)
SCRIPT ERROR: Parse Error: Unexpected "Identifier" in class body.
          at: GDScript::reload (res://ScreenShaderTextureRect.gd:3)
SCRIPT ERROR: Parse Error: Unexpected "Identifier" in class body.
          at: GDScript::reload (res://util/Debug.gd:3)
SCRIPT ERROR: Parse Error: Unexpected "Identifier" in class body.
          at: GDScript::reload (res://player/player.gd:9)
SCRIPT ERROR: Parse Error: Expected end of statement after variable declaration, found "Identifier" instead.
          at: GDScript::reload (res://player/AnimatedSprite.gd:3)
SCRIPT ERROR: Parse Error: Unexpected "Identifier" in class body.
          at: GDScript::reload (res://player/PlayerDebug.gd:3)
SCRIPT ERROR: Parse Error: Unexpected "Identifier" in class body.
          at: GDScript::reload (res://player/UILifeBar.gd:3)
SCRIPT ERROR: Parse Error: Unexpected "Identifier" in class body.
          at: GDScript::reload (res://player/PlayerCrosshair.gd:3)
SCRIPT ERROR: Parse Error: Unexpected "Identifier" in class body.
          at: GDScript::reload (res://world/environment/background/ParallaxLayerClouds.gd:3)
SCRIPT ERROR: Parse Error: Unexpected "Identifier" in class body.
          at: GDScript::reload (res://Main2D.gd:3)
SCRIPT ERROR: Parse Error: Unexpected "Identifier" in class body.
          at: GDScript::reload (res://world/tilemaps/TileSet.gd:1)
ERROR: Cannot create tile. The tile is outside the texture or tiles are already present in the space the tile would cover.
   at: create_tile (scene/resources/tile_set.cpp:4016)
ERROR: The TileSetAtlasSource atlas has no tile at (0, 0).
   at: get_tile_data (scene/resources/tile_set.cpp:4425)

================================================================
handle_crash: Program crashed with signal 11
Engine version: Godot Engine v4.0.alpha13.official (59dcddc453ed5cde80da20cfbad92bb17953d503)
Dumping the backtrace. Please include this when reporting the bug to the project developer.
[1] /usr/lib/libc.so.6(+0x38a40) [0x7f8fad27fa40] (??:0)
[2] /home/user/opt/Godot_v4.0-alpha13_linux.64() [0x350a668] (??:0)
[3] /home/user/opt/Godot_v4.0-alpha13_linux.64() [0x3519b90] (??:0)
[4] /home/user/opt/Godot_v4.0-alpha13_linux.64() [0x351e886] (??:0)
[5] /home/user/opt/Godot_v4.0-alpha13_linux.64() [0x42efe49] (??:0)
[6] /home/user/opt/Godot_v4.0-alpha13_linux.64() [0x3395fdb] (??:0)
[7] /home/user/opt/Godot_v4.0-alpha13_linux.64() [0x4073be4] (??:0)
[8] /home/user/opt/Godot_v4.0-alpha13_linux.64() [0x4074339] (??:0)
[9] /home/user/opt/Godot_v4.0-alpha13_linux.64() [0x4079eef] (??:0)
[10] /home/user/opt/Godot_v4.0-alpha13_linux.64() [0x3393fac] (??:0)
[11] /home/user/opt/Godot_v4.0-alpha13_linux.64() [0x4073be4] (??:0)
[12] /home/user/opt/Godot_v4.0-alpha13_linux.64() [0x4074339] (??:0)
[13] /home/user/opt/Godot_v4.0-alpha13_linux.64() [0x4079eef] (??:0)
[14] /home/user/opt/Godot_v4.0-alpha13_linux.64() [0x3393fac] (??:0)
[15] /home/user/opt/Godot_v4.0-alpha13_linux.64() [0x4073be4] (??:0)
[16] /home/user/opt/Godot_v4.0-alpha13_linux.64() [0x4074339] (??:0)
[17] /home/user/opt/Godot_v4.0-alpha13_linux.64() [0x4079eef] (??:0)
[18] /home/user/opt/Godot_v4.0-alpha13_linux.64() [0x3393fac] (??:0)
[19] /home/user/opt/Godot_v4.0-alpha13_linux.64() [0x4073be4] (??:0)
[20] /home/user/opt/Godot_v4.0-alpha13_linux.64() [0x4074339] (??:0)
[21] /home/user/opt/Godot_v4.0-alpha13_linux.64() [0x4079eef] (??:0)
[22] /home/user/opt/Godot_v4.0-alpha13_linux.64() [0x486aae9] (??:0)
[23] /home/user/opt/Godot_v4.0-alpha13_linux.64() [0x1cc9f1c] (??:0)
[24] /home/user/opt/Godot_v4.0-alpha13_linux.64() [0x42edf4d] (??:0)
[25] /home/user/opt/Godot_v4.0-alpha13_linux.64() [0x1bb580a] (??:0)
[26] /home/user/opt/Godot_v4.0-alpha13_linux.64() [0x42888d4] (??:0)
[27] /home/user/opt/Godot_v4.0-alpha13_linux.64() [0x2a053fe] (??:0)
[28] /home/user/opt/Godot_v4.0-alpha13_linux.64() [0x2a56c08] (??:0)
[29] /home/user/opt/Godot_v4.0-alpha13_linux.64() [0xe54386] (??:0)
[30] /home/user/opt/Godot_v4.0-alpha13_linux.64() [0xdeada9] (??:0)
[31] /usr/lib/libc.so.6(+0x232d0) [0x7f8fad26a2d0] (??:0)
[32] /usr/lib/libc.so.6(__libc_start_main+0x8a) [0x7f8fad26a38a] (??:0)
[33] /home/user/opt/Godot_v4.0-alpha13_linux.64() [0xe060de] (??:0)
-- END OF BACKTRACE --
================================================================
```

Perhaps it's an issue with my TileMaps and TileSets, I understand a bit has changed with them, however I thought it was mostly in regards to the poor UX in 3.4 (which is why I'm attempting to upgrade).

Is there an upgrade path from the older tilemaps/sets or will I have to manually dissect my project and port them?

### Steps to reproduce

See above.

**Comment by blipk** [comment]
Looks like the problem was this part:
`ERROR: Cannot create tile. The tile is outside the texture or tiles are already present in the space the tile would cover.`

After removing all the regions from my tilesets (some which where overlapping), the project now loads and the editor seems to be running fine, albeit my tilesets are completely empty, so I'm guessing theres a lot of breaking changes here.

I guess overlapping tiles aren't supported in the new version.

**Comment by fishclaws** [comment]
can I get more information on how to fix this in my project?

**Comment by Zireael07** [comment]
@fishclaws Read the comment above yours (remove all the regions or make sure they don't overlap)

**Comment by blipk** [comment]
@fishclaws 
To fix the error I had to open the project in 3.0 and delete all the regions/areas in the tileset, then import it into 4.0
There were many changes to the new 4.0 TileSets though, you really have to recreate everything.
